Automattic/wp-calypso
# Reader: Move reader-ui action types to a separate location

#### Changes proposed in this Pull Request

Currently, the reader UI action types are declared either in the main `action-types` file, or the `reader` one. Also, the `reader-ui` state tree is (intentionally) completely separate from `reader`. But by importing just a few action types from the reader action types file, we pull the rest of the reader action types into the main entry chunk, and that's undesired.

This PR moves the `reader-ui` action types in their own file. That will no longer pull the `reader` action types into the main entry point, which should have some positive impact on the its bundle size.

A subsequent optimization could be reconsidering how critical `requestUnseenStatus()` is at boot time. Right now, we request it as early as possible at boot time, while we could actually do it later, triggered by the relevant masterbar item once it has rendered.

#### Testing instructions

* Verify Calypso still builds properly.
* Smoke test reader UI (sidebar, seen posts, expanding and collapsing) and verify it still works well. 
* Verify all tests pass.
